### ISSUE TITLE
Possible Arduino DUE support?

### ISSUE THREAD

**Issue by SchmidtFranz** [jmriego/Fino#3]
Hey jmriego would it be possible for you to add support for the arduino due?
At the moment i cant get it to work. I tried some changes from the YukMingLaw pull request that adds some things to make the due work -> it now compiles without errors but it doesn't show up as a USB Device.
Without the changes there are some errors preventing it from compiling.

If you can help me with this, that would be great.

**Comment by SchmidtFranz** [comment]
I'm going to try to make it work again also (if I get it working I'm putting in a pull request). It's not that important at the moment but in the future it would be nice, because on the Leonardo I'm limited by ram (I use about 75% atm but so far it's fine) and the extra serial ports on the due would be nice.

**Comment by jmriego** [comment]
this repo is based on ArduinoJoystickWithFFBLibrary which is based on https://github.com/MHeironimus/ArduinoJoystickLibrary

It seems like that last repo already supports the DUE! I'm sure it's possible to reuse their code and add support here

**Comment by SchmidtFranz** [comment]
That was my next idea, but so far what I tested sending works but receiving data (for ffb) is not implemented there, but I'm going to try again, next week, it can't be that difficult.

**Comment by SchmidtFranz** [comment]
If got a bit of an off topic question, how is the force strength and "precision" of the brunner compared to the msffb2. I can't find much about it on the internet.

**Comment by jmriego** [comment]
the strength and precision are much better on the Brunner base! Mind you that I have read some comments that the forces were too weak by some people who used it with a Thrustmaster grip but that thing is really heavy. I think it was over 1 kg.
You're better off by getting a different grip I think. I have a Virpil and I'm really happy with this.
The only thing that I don't like compared to the msffb2 is that there's always some friction when the device is on. Just imagine that you always had a friction effect on a low setting. I'm too used to the feel of stalls in msffb2 and don't feel that as easily on the Brunner

**Comment by jmriego** [comment]
hi @SchmidtFranz ! 

I just spent some time fixing the compilation for Arduino Due. I don't have an Arduino Due so I can't actually test this but let me know if this works for you!

**Comment by SchmidtFranz** [comment]
Thanks @jmriego for looking into this, unfortunately i dont have much time at the moment, but i got my new stick with hoverboard motors almost working now with a arduino leonardo. If you are interested in this i can send you a video of it when its finished.
if you want to test the arduino due yourself i would be willing to buy you one so you can test it yourself. or i test it in a month when i have time again and report back to you.

**Comment by jmriego** [comment]
no worries @SchmidtFranz . Of course if you send me or lend me an Arduino Due I'll look into it but if you prefer to wait that's also fine!

**Comment by jmriego** [comment]
I recently got a Due and I'm starting to test this. Sadly it seems like it's not working properly. The Joystick part works fine but the Force Feedback doesn't.
From what I've gathered online, the Arduino Due USB libraries are not in a great state so I have to do some debugging and figure out what the issue is. At the moment it's possible to send info from the Due to the host, but not the other way around. There's plenty of documentation online so this should be fixable even though I cannot find an example of anyone who has done this kind of stuff before
